destinations/wavefront/handler.js

```diff
@@ -0,0 +1,133 @@
+
+// This example hits the WAVEFRONT REST API V2 events endpoint
+// Note 1: The wavefront instance name is different for each customer
+// Note 2: v2 is the only supported version as of 1/15/2020
+
+//TODO: Move these to settings once more than apiKey is supported
+let wavefrontInstance = "longboard";
+let wavefrontApiVersion = "v2";
+let wavefrontResource = "event";
+
+//Create endpoint
+const endpoint = `https://${wavefrontInstance}.wavefront.com/api/${wavefrontApiVersion}/${wavefrontResource}`;
+const url = new URL(endpoint);
+
+//Properties we need in the track event
+//Note 1: we only cover instantaneous events for now as ending events later is only possible via wavefront UI manually
+//Note 2: startTime and endTime are just set to timestamp
+//TODO: Convert some important context properties to tags
+const annotationProperties = [
+    { key: "type", validateFunc: (input) => { return _.isString(input)} },
+    { key: "details", validateFunc: (input) => { return _.isString(input)} },
+    { key: "severity", validateFunc: (input) => { return _.isString(input) && _.indexOf(["info", "warn", "severe"], input) > -1 } }];
+
+/**
+ * @param {SpecTrack} event The track event
+ * @param {Object.<string, any>} settings Custom settings
+ * @return void
+ */
+async function onTrack(event, settings) {
+  let { properties, timestamp } = event;
+
+  let eventName = event.event;
+  let timestampInSeconds = Date.parse(timestamp);
+  let annotations = {
+      prettyName: eventName
+  };
+  annotationProperties.forEach(subproperty => {
+    let { key, validateFunc } = subproperty;
+    if(properties.hasOwnProperty(key) && validateFunc(properties[key]))
+      annotations[key] = properties[key];
+  });
+
+  let body = {
+    name: eventName,
+    startTime: timestampInSeconds,
+    endTime: timestampInSeconds + 1,
+    annotations
+  };
+
+  //TODO: Is there a reason to create URL object and then stringify again?
```

Comment:
You don't need to create URL then stringify. (This line and 12-13.)

```
const endpoint = `https://${wavefrontInstance}.wavefront.com/api/${wavefrontApiVersion}/${wavefrontResource}`;
const url = new URL(endpoint);
```

Can be 
```
const url = new URL(`https://${wavefrontInstance}.wavefront.com/api/${wavefrontApiVersion}/${wavefrontResource}`);
```


and used like this: 
```
const res = await fetch(url.toString(), {
...etc.```

Reply:
@kellymtrinh made this change as well: https://github.com/segmentio/functions-library/pull/33/commits/49ad7606caeb04bead1f47addbdb1c134894ae89